ccl/backupccl: TestCleanupIntentsDuringBackupPerformanceRegression failed

ccl/backupccl.TestCleanupIntentsDuringBackupPerformanceRegression [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9330001?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9330001?buildTab=artifacts#/) on master @ [d671fad8a0ca13e2ff2ed4c5f7c8a596bb0b8b33](https://github.com/cockroachdb/cockroach/commits/d671fad8a0ca13e2ff2ed4c5f7c8a596bb0b8b33):


```
=== RUN   TestCleanupIntentsDuringBackupPerformanceRegression
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestCleanupIntentsDuringBackupPerformanceRegression126413690
    test_log_scope.go:79: use -show-logs to present logs inline
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
    backup_intents_test.go:83: Backup took 20.545279836s
    backup_intents_test.go:84: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl_test/pkg/ccl/backupccl/backup_intents_test.go:84
        	Error:      	Max difference between 2023-03-30 05:19:05.203122049 +0000 UTC m=+85.639147501 and 2023-03-30 05:18:44.657842271 +0000 UTC m=+65.093867665 allowed is 20s, but difference was 20.545279836s
        	Test:       	TestCleanupIntentsDuringBackupPerformanceRegression
        	Messages:   	Time to make backup
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestCleanupIntentsDuringBackupPerformanceRegression126413690
--- FAIL: TestCleanupIntentsDuringBackupPerformanceRegression (77.52s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCleanupIntentsDuringBackupPerformanceRegression.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26188

ccl/backupccl.TestCleanupIntentsDuringBackupPerformanceRegression [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9721442?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9721442?buildTab=artifacts#/) on master @ [8124bff03d073f35f4d2b6a2048c7f4417d757d9](https://github.com/cockroachdb/cockroach/commits/8124bff03d073f35f4d2b6a2048c7f4417d757d9):


```
=== RUN   TestCleanupIntentsDuringBackupPerformanceRegression
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestCleanupIntentsDuringBackupPerformanceRegression4250419719
    test_log_scope.go:79: use -show-logs to present logs inline
    test_server_shim.go:357: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
    backup_intents_test.go:83: Backup took 22.322219661s
    backup_intents_test.go:84: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl_test/pkg/ccl/backupccl/backup_intents_test.go:84
        	Error:      	Max difference between 2023-04-21 05:21:32.941791751 +0000 UTC m=+112.546840856 and 2023-04-21 05:21:10.619572138 +0000 UTC m=+90.224621195 allowed is 20s, but difference was 22.322219661s
        	Test:       	TestCleanupIntentsDuringBackupPerformanceRegression
        	Messages:   	Time to make backup
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestCleanupIntentsDuringBackupPerformanceRegression4250419719
--- FAIL: TestCleanupIntentsDuringBackupPerformanceRegression (101.17s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100268 ccl/backupccl: TestCleanupIntentsDuringBackupPerformanceRegression failed [C-test-failure O-robot T-disaster-recovery branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCleanupIntentsDuringBackupPerformanceRegression.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestCleanupIntentsDuringBackupPerformanceRegression [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9750456?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9750456?buildTab=artifacts#/) on master @ [23ea65e4b5cfaa5e02298cddfbe8f9fbe581456b](https://github.com/cockroachdb/cockroach/commits/23ea65e4b5cfaa5e02298cddfbe8f9fbe581456b):


```
=== RUN   TestCleanupIntentsDuringBackupPerformanceRegression
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestCleanupIntentsDuringBackupPerformanceRegression305568542
    test_log_scope.go:79: use -show-logs to present logs inline
    backup_intents_test.go:83: Backup took 22.113421581s
    backup_intents_test.go:84: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl_test/pkg/ccl/backupccl/backup_intents_test.go:84
        	Error:      	Max difference between 2023-04-23 05:58:02.048502077 +0000 UTC m=+92.471026231 and 2023-04-23 05:57:39.93508048 +0000 UTC m=+70.357604650 allowed is 20s, but difference was 22.113421581s
        	Test:       	TestCleanupIntentsDuringBackupPerformanceRegression
        	Messages:   	Time to make backup
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestCleanupIntentsDuringBackupPerformanceRegression305568542
--- FAIL: TestCleanupIntentsDuringBackupPerformanceRegression (80.36s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100268 ccl/backupccl: TestCleanupIntentsDuringBackupPerformanceRegression failed [C-test-failure O-robot T-disaster-recovery branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCleanupIntentsDuringBackupPerformanceRegression.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestCleanupIntentsDuringBackupPerformanceRegression [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9804206?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9804206?buildTab=artifacts#/) on master @ [2f4231084e27760b4927e15da9b6d46705dc8552](https://github.com/cockroachdb/cockroach/commits/2f4231084e27760b4927e15da9b6d46705dc8552):


```
=== RUN   TestCleanupIntentsDuringBackupPerformanceRegression
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestCleanupIntentsDuringBackupPerformanceRegression3705290142
    test_log_scope.go:79: use -show-logs to present logs inline
    test_server_shim.go:357: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
    backup_intents_test.go:83: Backup took 22.497673429s
    backup_intents_test.go:84: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl_test/pkg/ccl/backupccl/backup_intents_test.go:84
        	Error:      	Max difference between 2023-04-26 05:22:58.254425748 +0000 UTC m=+105.431106711 and 2023-04-26 05:22:35.756752386 +0000 UTC m=+82.933433282 allowed is 20s, but difference was 22.497673429s
        	Test:       	TestCleanupIntentsDuringBackupPerformanceRegression
        	Messages:   	Time to make backup
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestCleanupIntentsDuringBackupPerformanceRegression3705290142
--- FAIL: TestCleanupIntentsDuringBackupPerformanceRegression (94.93s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100268 ccl/backupccl: TestCleanupIntentsDuringBackupPerformanceRegression failed [C-test-failure O-robot T-disaster-recovery branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCleanupIntentsDuringBackupPerformanceRegression.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/backup_intents_test.go
+++ b/pkg/ccl/backupccl/backup_intents_test.go
@@ -31,7 +31,8 @@ func TestCleanupIntentsDuringBackupPerformanceRegression(t *testing.T) {
 	defer log.Scope(t).Close(t)
 	defer ccl.TestingEnableEnterprise()()
 
-	skip.UnderRace(t, "measures backup times not to regress, can't work under race")
+	skip.UnderRace(t, "measures backup times to ensure intent resolution does not regress, can't work under race")
+	skip.UnderDeadlock(t, "measures backup times to ensure intent resolution does not regress, can't work under deadlock")
 
 	// Time to create backup in presence of intents differs roughly 10x so some
 	// arbitrary number is picked which is 2x higher than current backup time on

